prow/cmd/crier/README.md

```diff
@@ -0,0 +1,57 @@
+# Crier
+
+Crier reports your prowjobs on their status changes.
+
+## Reporter Controller
+
+Crier supports multiple reporters, each reporter will become a crier controller. Controllers
+will get prowjob change notifications from a shared informer, and process on a shared rate-limit
+queue. 
+
+If you are interested in how client-go works under the hood, the details are explained 
+[in this doc](https://github.com/kubernetes/sample-controller/blob/master/docs/controller-client-go.md)
+
+## Report client interface
+
+Each report client will implement the following interface:
+```go
+type reportClient interface {
+	Report(pj *v1.ProwJob) error
+	GetName() string
+	ShouldReport(pj *v1.ProwJob) bool
+}
+```
+
+`GetName` will return the name of your reporter, the name will be used as a key when we store previous
+reported state for each prowjob.
+
+`ShouldReport` will return if a prowjob should be handled by current reporter.
+
+`Report` is the actual report logic happens. Return `nil` means report is successful, and the reported
+state will be saved in the prowjob. Return an actual error if report fails, crier will re-add the prowjob
+key to the shared cache and retry up to 5 times.
+
+## Existing available reporters
+
+### [Pubsub reporter](/prow/pubsub/reporter)
+
+You can enable pubsub reporter in crier by specifying `--pubsub` flag.
+
+You need to specify following labels in order for pubsub reporter to report your prowjob:
+
+| Label                          | Description                                                                        |
+| ------------------------------ | -----------------------------------------------------------------------------------|
+| `"prow.k8s.io/pubsub-project"` | your gcp project where pubsub channel lives                                        |
+| `"prow.k8s.io/pubsub-topic"`   | the [topic](https://cloud.google.com/pubsub/docs/publisher) of your pubsub message |
+| `"prow.k8s.io/pubsub-runID"`   | your prowjob ID to be published                                                    |
```

Comment:
The description of pubsub runID is:
A user assigned job id. It's tied to the prowjob, serves as a name tag and help user to differentiate results in multiple pubsub messages.

Reply:
:+1: done